capirca/lib/nsxt.py

```diff
@@ -0,0 +1,339 @@
+
+  def __init__(self, term, filter_type, applied_to=None, af=4):
+    self.term = term
+    # Our caller should have already verified the address family.
+    assert af in (4, 6)
+    self.af = af
+    self.filter_type = filter_type
+    self.applied_to = applied_to
+
+  def __str__(self):
+    """Convert term to a rule string.
+
+    Returns:
+      A rule as a string.
+
+    """
+    # Verify platform specific terms. Skip whole term if platform does not
+    # match.
+    if self.term.platform:
+      if 'nsxt' not in self.term.platform:
+        return ''
+    if self.term.platform_exclude:
+      if 'nsxt' in self.term.platform_exclude:
+        return ''
+
+    term_keywords = self.term.__dict__
+    unsupported_keywords = []
+    for key in term_keywords:
+      if term_keywords[key]:
+        if ('translated' not in key) and (key not in _NSXT_SUPPORTED_KEYWORDS):
+              unsupported_keywords.append(key)
+    if unsupported_keywords:
+      logging.warning('WARNING: The keywords %s in Term %s are not supported '
+                      'in Nsxt ', unsupported_keywords, self.term.name)
+
+    name = self.term.name
+
+    notes = ''
+    if self.term.comment:
+      notes = self.term.comment
+
+    action = 'ALLOW'
+    if self.term.action:
+      action = self.term.action
+
+    source_address = 'ANY'
+    if self.term.source_address:
+      source_address = []
+      for i in self.term.source_address:
+        source_address.append(str(i))
+
+    destination_address = 'ANY'
+    if self.term.destination_address:
+      destination_address = []
+      for i in self.term.destination_address:
+        destination_address.append(str(i))
+
+    services = 'ANY'
+    # if self.term.services:
+    #   services = self.term.services
+
+    profiles = 'ANY'
+    # if self.term.profiles:
+    #   profiles = self.term.profiles
+
+    scope = 'ANY'
+    # if self.term.scope:
+    #   scope = self.term.scope
+
+    direction = 'IN_OUT'
+    # if self.term.direction:
+    #   direction = self.term.direction
+
+    tag = ''
+    # if self.term.tag:
+    #   tag = self.term.tag
+
+    ip_protocol = ''
+    # if self.term.ip_protocol:
+    #   ip_protocol = self.term.ip_protocol
+
+    rule =  {
+      "action": action,
+      "resource_type": "Rule",
+      "display_name": name,
+      "source_groups": source_address,
+      "destination_groups": destination_address,
+      "services": [
+          services
+      ],
+      "profiles": [
+          profiles
+      ],
+      "scope": [
+          scope
+      ],
+      "notes": notes,
+      "direction": direction,
+      "tag": tag,
+      "ip_protocol": ip_protocol
+    }
+
+    # ret_str = []
+
+    # ret_lines = []
+    # ret_lines.append('%s' % (rule))
+
+    # remove any trailing spaces and replace multiple spaces with singles
+    # stripped_ret_lines = [re.sub(r'\s+', ' ', x).rstrip() for x in ret_lines]
+    # ret_str.extend(stripped_ret_lines)
+
+    print(rule)
```

Comment:
Let's not leave a debug print to stdout when submitting the code. Check for correctness should be done using a test.